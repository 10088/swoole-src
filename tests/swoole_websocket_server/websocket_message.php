<?php
require_once __DIR__ . "/../include/swoole.inc";
include __DIR__ . "/../include/lib/class.websocket_client.php";

function start_swoole_ws_server()
{
    swoole_php_fork(function ()
    {
        $serv = new swoole_websocket_server("127.0.0.1", 9501);
        $serv->set(['log_file' => '/dev/null']);
        $serv->on('Open', function ($swoole_server, $req)
        {
        });
        $serv->on('Message', function ($swoole_server, $frame)
        {
            $swoole_server->push($frame->fd, "hello " . $frame->data);
            if ($frame->data == "shutdown")
            {
                $swoole_server->shutdown();
            }
        });
        $serv->on('Close', function ($swoole_server, $fd)
        {
        });
        $serv->start();
    });
}

sleep(1);    //wait the release of port 9501
start_swoole_ws_server();
sleep(1);

$cli = new WebsocketClient;
$connected = $cli->connect('127.0.0.1', 9501, '/');
assert($connected);
assert($cli->sendRecv("batman") == 'hello batman');
assert($cli->sendRecv("spiderman") == 'hello spiderman');
?>

