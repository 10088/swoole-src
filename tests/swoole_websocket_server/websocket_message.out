Warning: swoole_client::recv(): recv() failed. Error: Resource temporarily unavailable [11] in /src/swoole-src/tests/include/lib/class.websocket_client.php on line 51

Warning: assert(): assert($cli->sendRecv('batman') == 'hello batman') failed in /src/swoole-src/tests/swoole_websocket_server/websocket_message.php on line 36